src/protocol/SSLWrapper.cc

```diff
@@ -30,7 +29,8 @@ int SSLWrapper::encode(struct iovec vectors[], int max)
 {
 	BIO *bio = SSL_get_wbio(this->ssl);
 	struct iovec *iov;
-	void *buf;
+	char *ptr;
+	long len;
 	int ret;
 
 	ret = this->msg->encode(vectors, max);
```

Comment:
use ret with more than one meaning lead to bugs, see the following for loop (vectors + ret)